site/app/libraries/database/DatabaseQueries.php

```diff
@@ -728,25 +728,65 @@ public function toggleLikes(int $post_id, string $current_user): array {
             $inDatabase = isset($this->course_db->rows()[0]);
+
+            $sqlFilteredCount = "SELECT COUNT(*) AS filtered_likes_count
+                             FROM forum_upducks f
+                             JOIN users u ON f.user_id = u.user_id
+                             WHERE f.post_id = ?
+                             AND u.user_group IN (1, 2, 3)";
             if ($inDatabase) {
                 $this->course_db->query("DELETE FROM forum_upducks WHERE post_id = ? AND user_id = ?", [$post_id, $current_user]);
+                $action = "unlike";
                 $this->course_db->query("SELECT COUNT(*) AS likes_count FROM forum_upducks WHERE post_id = ?", [$post_id]);
                 $likesCount = intval($this->course_db->rows()[0]['likes_count']);
-                return ['unlike',$likesCount];
+                $this->course_db->query($sqlFilteredCount, [$post_id]);
+                $filteredLikesCount = intval($this->course_db->rows()[0]['filtered_likes_count']);
             }
             else {
                 $this->course_db->query("INSERT INTO forum_upducks (post_id, user_id) VALUES (?, ?)", [$post_id, $current_user]);
+                $action = "like";
                 $this->course_db->query("SELECT COUNT(*) AS likes_count FROM forum_upducks WHERE post_id = ?", [$post_id]);
                 $likesCount = intval($this->course_db->rows()[0]['likes_count']);
-                return ['like',$likesCount];
+                $this->course_db->query($sqlFilteredCount, [$post_id]);
+                $filteredLikesCount = intval($this->course_db->rows()[0]['filtered_likes_count']);
             }
+            return [$action, $likesCount, $filteredLikesCount];
         }
         catch (DatabaseException $dbException) {
             if ($this->course_db->inTransaction()) {
                 $this->course_db->rollback();
             }
-            return ["false",0];
+            return ["false",0,0];
+        }
+    }
+
+    /**
+     * @param int[] $post_ids
+     * @return int[]
+     */
+    public function getInstructorUpduck(array $post_ids): array {
+        if (count($post_ids) === 0) {
+            return [];
         }
+        $placeholders = $this->createParameterList(count($post_ids));
+
+        // SQL to join the forum_upducks and users tables, filter by user_group, and check against provided post_ids
+        $sql = "SELECT f.post_id
+                FROM forum_upducks f
+                JOIN users u ON f.user_id = u.user_id
+                WHERE f.post_id IN {$placeholders}
+                AND u.user_group IN (1, 2, 3)
+                GROUP BY f.post_id";
+
+        // Execute the query with the post_ids as parameters
+        $this->course_db->query($sql, $post_ids);
+        $result = [];
+
+        // Fetch the rows and store the post_id in the result array
+        foreach ($this->course_db->rows() as $row) {
+            array_push($result, $row['post_id']);
```

Comment:
It would be better to use PHP's array shorthand for consistency with the rest of the codebase:
```suggestion
            $result[] = $row['post_id'];
```